src/Histogram.ts

```diff
@@ -414,15 +449,15 @@ export default class Histogram {
 
   // @param {Object[]} controlPoints - array of {x:number 0..255, opacity:number 0..1, color:array of 3 numbers 0..255}
   // @return {Uint8Array} array of length 256*4 representing the rgba values of the gradient
-  lutGenerator_fromControlPoints(controlPoints) {
+  lutGenerator_fromControlPoints(controlPoints: ControlPoint[]): Lut {
     const lut = new Uint8Array(LUT_ARRAY_LENGTH).fill(0);
 
     if (controlPoints.length === 0) {
       return { lut: lut, controlPoints: controlPoints };
     }
 
     // ensure they are sorted in ascending order of x
-    controlPoints.sort((a, b) => (a.x > b.x ? 1 : -1));
+    controlPoints.sort((a, b) => (a.x > b.x ? 1 : a.x < b.x ? -1 : 0));
```

Comment:
Would `=> (a.x - b.x)` work here for the compare function?

Reply:
I believe it would.  This is a place that is actually potentially broken when you try to make a step function and pass it in.
The step function has two values at the same x and I'm not clear whether the sort is swapping their order or not. (The underlying implementation of sort is unknown to me; maybe it's documented somewhere)

Reply:
ha! it is documented:
```
Note: The ECMAScript Standard, 10th edition (2019) algorithm mandates stable sorting, which means elements that compare equal must remain in their original order with respect to each other. This behaviour may not be respected by older browsers.
```

Reply:
So... should we use `a.x - b.x` here for readability?

Reply:
No prob, will make that change!